Signal/src/ViewControllers/PhotoLibrary/ImagePickerController.swift

```diff
@@ -92,16 +107,38 @@ class ImagePickerGridController: UICollectionViewController, PhotoLibraryDelegat
         let scale = UIScreen.main.scale
         let cellSize = collectionViewFlowLayout.itemSize
         photoMediaSize.thumbnailSize = CGSize(width: cellSize.width * scale, height: cellSize.height * scale)
+
+        reloadDataAndRestoreSelection()
+    }
+
+    private func reloadDataAndRestoreSelection() {
+        guard let collectionView = collectionView else {
+            owsFailDebug("Missing collectionView.")
+            return
+        }
+
+        collectionView.reloadData()
+        collectionView.layoutIfNeeded()
```

Comment:
`reloadData` clears the selection, so we need to restore it.

Reply:
I think at some point in the commit we have gone from default scroll position being the last item in the collection to the first. This means the user would now have to scroll down through potentially thousands of images to get to the most recent. Any idea where this might have occurred?